test/common/instance_api.test.lua

```diff
@@ -0,0 +1,131 @@
+test:is(instance_info.curr_conns, 0, 'default value of curr_conns is correct')
+test:is(instance_info.total_items, 0, 'default value of total_items is correct')
+test:is(instance_info.evictions, 0, 'default value of evictions is correct')
+test:is(instance_info.auth_errors, 0, 'default value of auth_errors is correct')
+test:is(instance_info.auth_cmds, 0, 'default value of auth_cmds is correct')
+test:is(instance_info.reclaimed, 0, 'default value of reclaimed is correct')
+test:is(instance_info.cmd_incr, 0, 'default value of cmd_incr is correct')
+test:is(instance_info.cmd_decr, 0, 'default value of cmd_decr is correct')
+test:is(instance_info.cmd_touch, 0, 'default value of cmd_touch is correct')
+test:is(instance_info.touch_hits, 0, 'default value of touch_hits is correct')
+test:is(instance_info.touch_misses, 0, 'default value of touch_misses is correct')
+test:is(instance_info.decr_misses, 0, 'default value of decr_misses is correct')
+test:is(instance_info.curr_items, 0, 'default value of curr_items is correct')
+test:is(instance_info.total_conns, 0, 'default value of total_conns is correct')
+test:is(instance_info.incr_misses, 0, 'default value of incr_misses is correct')
+test:is(instance_info.cmd_flush, 0, 'default value of cmd_flush is correct')
+test:is(instance_info.decr_hits, 0, 'default value of decr_hits is correct')
+test:is(instance_info.incr_hits, 0, 'default value of incr_hits is correct')
+test:is(instance_info.cmd_delete, 0, 'default value of cmd_delete is correct')
+test:is(instance_info.cas_misses, 0, 'default value of cas_misses is correct')
+test:is(instance_info.bytes_read, 0, 'default value of bytes_read is correct')
+test:is(instance_info.cas_badval, 0, 'default value of cas_badval is correct')
+test:is(instance_info.delete_hits, 0, 'default value of delete_hits is correct')
+test:is(instance_info.cmd_get, 0, 'default value of cmd_get is correct')
+test:is(instance_info.cas_hits, 0, 'default value of cas_hits is correct')
+test:is(instance_info.cmd_set, 0, 'default value of cmd_set is correct')
+test:is(instance_info.get_misses, 0, 'default value of get_misses is correct')
+
+-- instance:cfg()
+
+local updated_expire_full_scan_time = mc.opts.expire_full_scan_time + 100
+local updated_expire_items_per_iter = mc.opts.expire_items_per_iter + 50
+local updated_expire_enabled = not mc.opts.expire_enabled
+local updated_protocol = 'binary'
+local updated_name = mc.opts.name .. '_hello'
+local updated_verbosity = mc.opts.verbosity + 1
+local updated_sasl = not mc.opts.sasl
+
+local self_instance = mc:cfg({
+    expire_full_scan_time = updated_expire_full_scan_time,
+    expire_items_per_iter = updated_expire_items_per_iter,
+    expire_enabled = updated_expire_enabled,
+    protocol = updated_protocol,
+    name = updated_name,
+    verbosity = updated_verbosity,
+    sasl = updated_sasl,
+})
+test:isnt(self_instance, nil, 'instance:cfg() returns a self object')
```

Comment:
I didn't quite understand from the README: isn't `self_instance == mc`?

Reply:
Yes, according to documented behavior in the README mc is the same as self_instance. assert checks that behavior.
See also an implementation of stop() method: https://github.com/tarantool/memcached/blob/master/memcached/init.lua#L310-L320

Reply:
Wouldn't this kind of test be better?
```suggestion
test:is(self_instance, mc, 'instance:stop() returns self object')
```

Reply:
Fixed.